packages/server/src/joi.ts

```diff
@@ -0,0 +1,89 @@
+import Joi from "@hapi/joi";
+import { courseRoutes } from "./api/courseRoutes";
+import { QuestionType } from "@template/common";
+
+// type CoursePartial
+export const CoursePartialSchema = Joi.object({
```

Comment:
for all of the Joi stuff, do we want to let things like `id`, `name` be null? If not, you'll want to add `.options({presence: 'required'})` to each of these ala [here](https://hapi.dev/module/joi/api/?v=17.1.1#anyvalidatevalue-options)

Reply:
I think that the `schema: CourseSchema.options({ presence: "required" }),` in the courseRoutes.ts file bubbles up so it requires non-null for all the objects that it is composed of.